connection_test.go

```diff
@@ -63,13 +63,15 @@ func TestHostFunctions(t *testing.T) {
 	h = Host{
 		Connection: Connection{
 			SSH: &SSH{
-				Address: "10.0.0.1",
+				Address: "127.0.0.1",
 			},
 		},
 	}
 	require.NoError(t, defaults.Set(&h))
 	require.Equal(t, "SSH", h.Protocol())
-	require.Equal(t, "10.0.0.1", h.Address())
+	require.Equal(t, "127.0.0.1", h.Address())
+	require.NoError(t, h.Connect())
+	h.Disconnect()
```

Comment:
These will fail on any host that does not have a local sshd that accepts "root" without password.

Reply:
![image](https://user-images.githubusercontent.com/224971/184298984-354f3ef6-29b1-4278-9dfb-6e1d019cab51.png)

Reply:
Indeed it will. There are anyway many assumptions there.

Remove the `require` then, just doing playing connect would be ok? the idea being of triggering agent routines.

Reply:
maybe something like `if err != nil { t.Logf("connection failed: %v", err) }`

Reply:
(it will get rid of the lint error)

Reply:
Didn't see this comment :)